src/hapi/pipelines/database/food_security.py

```diff
@@ -23,138 +23,189 @@ def __init__(
-                        dict_of_lists_add(
-                            population_totals,
-                            (
-                                ipc_type,
-                                time_period_start,
-                                time_period_end,
-                            ),
-                            population_total,
+                    else:
+                        continue
+                    if subnational_level == "adminone":
+                        if not adminone:
+                            add_message(
+                                warnings,
+                                dataset_name,
+                                f"Admin 1: could not match {full_adm1name}!",
+                            )
+                            continue
+                        if not exact:
+                            name = self._adminone.pcode_to_name[adminone]
+                            if adminone_name in food_sec_config["adm1_errors"]:
+                                add_message(
+                                    errors,
+                                    dataset_name,
+                                    f"Admin 1: ignoring erroneous {full_adm1name} match to {name} {(adminone)}!",
+                                )
+                                continue
+                            add_message(
+                                warnings,
+                                dataset_name,
+                                f"Admin 1: matching {full_adm1name} to {name} {(adminone)}",
+                            )
+                        admin2_ref = self.get_admin2_ref(
+                            subnational_level, adminone, dataset_name, errors
                         )
-                        for ipc_phase in IPCPhase:
-                            population_in_phase = values[
-                                population_in_phase_columns[ipc_phase.value]
-                            ][admin_code][irow]
-                            if population_in_phase is None:
-                                population_in_phase = 0
-                            population_in_phase = int(population_in_phase)
-                            # Sum the phase population in each row to aggregate admin 1.5 to admin 1
-                            dict_of_lists_add(
-                                population_in_phases,
-                                (
-                                    ipc_phase.value,
-                                    ipc_type,
-                                    time_period_start,
-                                    time_period_end,
-                                ),
-                                population_in_phase,
+                    elif subnational_level == "admintwo":
+                        if countryiso3 in food_sec_config["adm1_only"]:
+                            continue
+                        admintwo_name = row["Area"]
+                        if adminone_name:
```

Comment:
You can just do `full_adm2name = f"{countryiso3}|{adminone_name or 'NOT GIVEN'}|{admintwo_name}"`

Reply:
Elegant!